### DISCUSSION TITLE
On-demand Revalidation is just Stale While Revalidate?

### DISCUSSION THREAD

**Discussion by Linushellberg** [vercel/next.js#50385] [App Router]
Hi,
I'm currently exploring the on-demand revalidation feature offered by Next.js and have a couple of questions I was hoping you could help clarify.

From what I understand, calling revalidateTag simply marks the page as stale, triggering a regeneration process only after a new request for this specific page is made - essentially implementing a "stale-while-revalidate" strategy. Could you confirm if this understanding is correct?

Furthermore, if the above is indeed the case, I'm wondering why there isn't an option to trigger an immediate regeneration, thus ensuring that the very next request receives a fresh version of the page. This seems particularly crucial in use-cases where serving the most up-to-date content is paramount.

Also, I'm trying to grasp the full benefit of static generation in this scenario. If static generation is used until the first revalidation, and post revalidation it behaves similarly to dynamic rendering with a "stale-while-revalidate" strategy at the CDN level, then does it mean that the primary advantage of static generation is speed for the first set of requests until revalidation is triggered?

Thank you for taking the time to read and respond to my queries. I am looking forward to your insight and feedback

**Comment by MihaiCGL** [comment]
I am asking myself the same questions. Up
